tests/contrib/molten/test_molten.py

```diff
@@ -0,0 +1,143 @@
+        self.assertEqual(response.json(), 'Hello 24 year old named Jim!')
+        span = spans[0]
+        self.assertEqual(span.service, self.TEST_SERVICE)
+        self.assertEqual(span.name, 'molten.request')
+        self.assertEqual(span.get_tag('http.method'), 'GET')
+        self.assertEqual(span.get_tag('http.url'), '/hello/Jim/24')
+        self.assertEqual(span.get_tag('http.status_code'), '200')
+
+    def test_resources(self):
+        """ Tests request has expected span resources """
+        response = molten_client()
+        spans = self.tracer.writer.pop()
+
+        expected_resources = [
+            'molten.app.__call__',
+            'molten.middleware.ResponseRendererMiddleware',
+            'molten.components.HeaderComponent',
+            'molten.components.CookiesComponent',
+            'molten.components.QueryParamComponent',
+            'molten.components.RequestBodyComponent',
+            'molten.components.RequestDataComponent',
+            'molten.components.SchemaComponent',
+            'molten.components.UploadedFileComponent',
+            'molten.components.HeaderComponent',
+            'molten.components.CookiesComponent',
+            'molten.components.QueryParamComponent',
+            'molten.components.RequestBodyComponent',
+            'molten.components.RequestDataComponent',
+            'molten.components.SchemaComponent',
+            'molten.components.UploadedFileComponent',
+            'GET /hello/{name}/{age}',
+            'molten.renderers.JSONRenderer'
+        ]
+
+        # Addition of `UploadedFileComponent` in 0.7.2 changes expected spans
+        if MOLTEN_VERSION >= (0, 7, 2):
+            self.assertEqual([s.resource for s in spans],
+                expected_resources)
+        else:
+            self.assertEqual([s.resource for s in spans],
+                [r for r in expected_resources if r != 'molten.components.UploadedFileComponent'])
+
+    def test_distributed_tracing(self):
+        """ Tests whether span IDs are propogated when distributed tracing is on """
+        def prepare_environ(environ):
+            environ.update({
+                'DD_MOLTEN_DISTRIBUTED_TRACING': 'True',
+                'HTTP_X_DATADOG_TRACE_ID': '100',
+                'HTTP_X_DATADOG_PARENT_ID': '42',
+            })
+            return environ
+
+        response = molten_client(prepare_environ=prepare_environ)
+        spans = self.tracer.writer.pop()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), 'Hello 24 year old named Jim!')
+        span = spans[0]
+        self.assertEqual(span.service, self.TEST_SERVICE)
+        self.assertEqual(span.name, 'molten.request')
+        self.assertEqual(span.trace_id, 100)
+        self.assertEqual(span.parent_id, 42)
+
+    def test_unpatch_patch(self):
+        """ Tests unpatch-patch cycle """
+        unpatch()
+        self.assertTrue(Pin.get_from(molten) is None)
```

Comment:
There is also `self.assertIsNone(Pin.get_from(molten))`

Basically the same as what you have, this is informative, no need to change anything here at all.